(Maangchi) Ingredients not being correctly scraped

**Pre-filing checks**

- [x] I have searched for open issues that report the same problem
- [x] I have checked that the bug affects the latest version of the library

**Context**
Certain recipes have the number of people being served contained within the same heading tag as the ingredients string (see links below). This may be the cause of the ingredients for these recipes not being correctly scraped, leading to a NoneType error when calling the scraper's ingredients method

**The URL of the recipe(s) that are not being scraped correctly**

- https://www.maangchi.com/recipe/budae-jjigae
- https://www.maangchi.com/recipe/gimbap
- https://www.maangchi.com/recipe/maeuntang


**The results you expect to see**

List of ingredients for the above recipes

**The results (including any Python error messages) that you are seeing**

```
Traceback (most recent call last):
  File "/path/to/file", line 14, in <module>
    print(scraper.ingredients())
          ^^^^^^^^^^^^^^^^^^^^^
  File "/path/site-packages/recipe_scrapers/plugins/exception_handling.py", line 63, in decorated_method_wrapper
    return decorated(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/path/site-packages/recipe_scrapers/plugins/html_tags_stripper.py", line 74, in decorated_method_wrapper
    decorated_func_result = decorated(self, *args, **kwargs)
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/path/site-packages/recipe_scrapers/plugins/schemaorg_fill.py", line 57, in decorated_method_wrapper
    return decorated(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/path/anaconda3/lib/python3.11/site-packages/recipe_scrapers/maangchi.py", line 11, in ingredients
    before = self.soup.find("h2", string="Ingredients").find_all_next("li")
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'find_all_next'
make: *** [run] Error 1
```

--- a/recipe_scrapers/maangchi.py
+++ b/recipe_scrapers/maangchi.py
@@ -1,3 +1,5 @@
+import re
+
 from ._abstract import AbstractScraper
 from ._utils import normalize_string
 
@@ -8,7 +10,9 @@ class Maangchi(AbstractScraper):
         return "maangchi.com"
 
     def ingredients(self):
-        before = self.soup.find("h2", string="Ingredients").find_all_next("li")
+        before = self.soup.find(
+            "h2", string=re.compile(r"Ingredients(\s*\(.*\))?")
+        ).find_all_next("li")
         after = self.soup.find("h2", string="Directions").find_all_previous("li")
         list_before = [normalize_string(b.get_text()) for b in before]
         list_after = [normalize_string(a.get_text()) for a in after]

--- a/recipe_scrapers/maangchi.py
+++ b/recipe_scrapers/maangchi.py
@@ -10,9 +10,9 @@ class Maangchi(AbstractScraper):
         return "maangchi.com"
 
     def ingredients(self):
-        before = self.soup.find(
-            "h2", string=re.compile(r"Ingredients(\s*\(.*\))?")
-        ).find_all_next("li")
+        before = self.soup.find("h2", string=re.compile(r"Ingredients")).find_all_next(
+            "li"
+        )
         after = self.soup.find("h2", string="Directions").find_all_previous("li")
         list_before = [normalize_string(b.get_text()) for b in before]
         list_after = [normalize_string(a.get_text()) for a in after]

